src/main/scala/is/hail/utils/HashMethods.scala

```diff
@@ -0,0 +1,20 @@
+package is.hail.utils
+
+// see e.g. Thorup, "High Speed Hashing for Integers and Strings", for explanation of "multiply-shift" hash functions,
+// both the universal and 2-independent constructions
+class UnivHash32(outBits: Int, factor: Int) extends (Int => Int) {
+  require(0 <= outBits && outBits <= 32)
+
+  val oddFactor = factor | 1
+  val shift: Int = 32 - outBits
```

Comment:
Storing these are going to be slower than just computing them (both ~1 instruction).  I'd make them `def`s (or inline them outright).